src/main/java/org/opensearch/knn/index/query/nativelib/NativeEngineKnnVectorQuery.java

```diff
@@ -121,12 +132,35 @@ private List<Map<Integer, Float>> doRescore(
                     .isParentHits(false)
                     .knnQuery(knnQuery)
                     .build();
-                return knnWeight.exactSearch(leafReaderContext, exactSearcherContext);
+                final Map<Integer, Float> docIdScoreMap = knnWeight.exactSearch(entry.getKey(), exactSearcherContext);
+                return new AbstractMap.SimpleEntry<>(entry.getKey(), docIdScoreMap);
             });
         }
         return indexSearcher.getTaskExecutor().invokeAll(rescoreTasks);
     }
 
+    private List<Map.Entry<LeafReaderContext, Map<Integer, Float>>> doExactSearch(
```

Comment:
doRescore and this method are more or less similar, any possiblity that we can reuse something here?